strawberry/experimental/pydantic/conversion.py

```diff
@@ -42,6 +45,10 @@ def _convert_from_pydantic_to_strawberry_type(
     elif is_scalar(type_):
         return data
     else:
+        # in the case of an interface, the concrete type may be more specific
+        # than the type in the field definition
```

Comment:
uhm, not sure I understand the comment here 🤔

Reply:
So in the case of an interface, the field definition will have some type, which is the base class of the hierarchy. If we just use the type definition to figure out which conversion function to call, we will always convert the subclass to that base type, losing any fields that are only present on the subclass. That's how I found this issue, my objects were getting converted from `PydanticSpecificClass` to `StrawberryBaseClass` instead of `StrawberrySpecificClass`.